etna/transforms/decomposition/deseasonal.py

```diff
@@ -0,0 +1,211 @@
+        ----------
+        df:
+            Features dataframe with time
+
+        Returns
+        -------
+        result: pd.DataFrame
+            Dataframe with extracted features
+        """
+        result = df
+        seasonal = self._roll_seasonal(result[self.in_column])
+        if self.model == "additive":
+            result[self.in_column] -= seasonal
+        else:
+            if np.any(result[self.in_column] <= 0):
+                raise ValueError(
+                    "The input column contains zero or negative values,"
+                    "but multiplicative seasonality can not work with such values."
+                )
+            result[self.in_column] /= seasonal
+        return result
+
+    def inverse_transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Add seasonal component.
+
+        Parameters
+        ----------
+        df:
+            Features dataframe with time
+
+        Returns
+        -------
+        result: pd.DataFrame
+            Dataframe with extracted features
+        """
+        result = df
+        seasonal = self._roll_seasonal(result[self.in_column])
+        if self.model == "additive":
+            result[self.in_column] += seasonal
+        else:
+            if np.any(result[self.in_column] <= 0):
+                raise ValueError(
+                    "The input column contains zero or negative values,"
+                    "but multiplicative seasonality can not work with such values."
+                )
+            result[self.in_column] *= seasonal
+        if self.in_column == "target":
+            quantiles = match_target_quantiles(set(result.columns))
+            for quantile_column_nm in quantiles:
+                if self.model == "additive":
+                    result.loc[:, quantile_column_nm] += seasonal
+                else:
+                    if np.any(result.loc[quantile_column_nm] <= 0):
+                        raise ValueError(
+                            f"The {quantile_column_nm} column contains zero or negative values,"
+                            "but multiplicative seasonality can not work with such values."
+                        )
+                    result.loc[:, quantile_column_nm] *= seasonal
+        return result
+
+
+class DeseasonalityTransform(ReversiblePerSegmentWrapper):
+    """Transform that uses :py:class:`statsmodels.tsa.seasonal.seasonal_decompose` to subtract season from the data.
+
+    Warning
+    -------
+    This transform can suffer from look-ahead bias. For transforming data at some timestamp
+    it uses information from the whole train part.
+    """
+
+    def __init__(self, in_column: str, period: int, model: str = "additive"):
+        """
+        Init DeseasonalityTransform.
+
+        Parameters
+        ----------
+        in_column:
+            name of processed column
+        period:
+            size of seasonality
+        model:
+            'additive' (default) or 'multiplicative'
```

Comment:
May be we should explain what does it do. May be we can add `Literal` types in model typing for more simple usage.